packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -169,9 +187,42 @@ - (instancetype)initWithCameraName:(NSString *)cameraName
   _motionManager = [[CMMotionManager alloc] init];
   [_motionManager startAccelerometerUpdates];
 
-  if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
-    return nil;
+  NSError *outError = nil;
+
+  if (_fps) {
+    // The frame rate can be changed only on a locked for configuration device.
+    if ([_captureDevice lockForConfiguration:&outError]) {
+      [_videoCaptureSession beginConfiguration];
+
+      // Possible values for presets are hard-coded in FLT interface having
+      // corresponding AVCaptureSessionPreset counterparts.
+      // If _resolutionPreset is not supported by camera there is
+      // fallback to lower resolution presets.
+      // If none can be selected there is error condition.
+      if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
+        [_videoCaptureSession commitConfiguration];
+        [_captureDevice unlockForConfiguration];
+        return nil;
+      }
+
+      _captureDevice.activeVideoMinFrameDuration = CMTimeMake(1, [_fps intValue]);
+      _captureDevice.activeVideoMaxFrameDuration = CMTimeMake(1, [_fps intValue]);
```

Comment:
If you are going to use the same CMTime twice, make it once as a local variable and then use that variable twice.

Reply:
extracted local variable.

Reply:
`CMTimeMake(10, floor(fps * 10))` (nom/denom) added to support non-integral values, like 29.9.